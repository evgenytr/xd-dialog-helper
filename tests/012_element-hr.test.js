const hr = require('../src/main').types.HR;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {
    }),
    close: jest.fn(() => {
    }),
    cancel: jest.fn(() => {
    }),
    registerElement: jest.fn(() => undefined)
};

const rendered = hr.render('dialog', {
    id: 'hr',
    htmlAttributes: {
        someAttr: 'hello'
    }
}, actions);

describe('HR Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();

            expect(hr.render('dialog', {id: 'noHTML'}, actions)).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return undefined as value', () => {
            expect(hr.value({wrapper: document.createElement('hr')})).toBeUndefined();
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(hr.valid({})).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(hr.type).toMatchSnapshot();
    });
});
