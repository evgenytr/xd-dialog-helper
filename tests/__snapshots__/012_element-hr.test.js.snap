// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Horizontal Rule"`;

exports[`HR Element render() should render correctly 1`] = `
Object {
  "input": undefined,
  "props": Object {
    "htmlAttributes": Object {
      "someAttr": "hello",
    },
    "id": "hr",
  },
  "type": Object {
    "render": [Function],
    "type": "Horizontal Rule",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <hr
    id="dialog-hr"
    someattr="hello"
  />,
}
`;

exports[`HR Element render() should render correctly 2`] = `
Object {
  "input": undefined,
  "props": Object {
    "id": "noHTML",
  },
  "type": Object {
    "render": [Function],
    "type": "Horizontal Rule",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <hr
    id="dialog-noHTML"
  />,
}
`;
